py-polars/polars/functions/range/date_range.py

```diff
@@ -126,42 +122,38 @@ def date_range(
     Notes
     -----
     1) If both ``start`` and ``end`` are passed as date types (not datetime), and the
-    interval granularity is no finer than 1d, the returned range is also of
-    type date. All other permutations return a datetime Series.
+       interval granularity is no finer than 1d, the returned range is also of
+       type date. All other permutations return a datetime Series.
 
-    2) Because different months of the year have differing numbers of days, the offset
-    strings "1mo" and "1y" are not well-defined units of time, and vary according to
-    their starting point. For example, February 1st offset by one month returns a time
-    28 days later (in a non-leap year), whereas May 1st offset by one month returns a
-    time 31 days later. In general, an offset of one month selects the same day in the
-    following month. However, this is not always intended: when one begins Febrary 28th
-    and offsets by 1 month, does the user intend to target March 28th (the next month
-    but same day), or March 31st (the end of the month)?
+       .. deprecated:: 0.19.3
 
-    Polars uses the first approach: February 28th offset by 1 month is March 28th. When
-    a date-series is generated, each date is offset as of the prior date, meaning that
-    if one began January 31st, 2023, and offset by ``1mo_saturating`` until May 31st,
-    the following dates would be generated:
+           In a future version of Polars, `date_range` will always return a `Date`.
+           Please use :func:`datetime_range` if you want a `Datetime` instead.
```

Comment:
can't get the link to show up correctly in sphinx 🤔

Reply:
Hm, try:
```
:func:`~polars.datetime_range`
```

Reply:
got it, `datetime_range*` were just missing from `functions.rst` (missed that when reviewing, apologies)

Reply:
Ah, thanks. Good catch!